exalearn-ui/src/app/components/speaking/speaking.component.ts

```diff
@@ -104,21 +111,29 @@ export class SpeakingComponent implements OnInit {
 
 	async pushAudioToCloudService(): Promise<void> {
 		const file = new File(this.chunks, 'recording.webm');
+		this.isDataAvailable = true;
 		this.isRecordReadySpinner = true;
-		await this.audioStorage.uploadAudio(file, environment.cloudSpeaking).then((url) => {
-			this.audioUrlCloud = url;
-			this.isRecordReadySpinner = false;
-			this.recording = false;
-			const speakingAnswer: testAnswer = {
-				id: 0,
-				passedTestId: this.testPassedId,
-				questionId: this.topic.id,
-				reportId: null,
-				answer: this.audioUrlCloud,
-				assessment: 0
-			};
-			this.submit.addData('speaking', speakingAnswer);
-		});
+		await this.audioStorage
+			.uploadAudio(file, environment.cloudSpeaking)
+			.then((url) => {
+				this.audioUrlCloud = url;
+				this.isRecordReadySpinner = false;
+				this.recording = false;
+				const speakingAnswer: testAnswer = {
```

Comment:
I think we can split this part to a separate method, which create necessary data object